lib/matplotlib/finance.py

```diff
@@ -1139,8 +1139,8 @@ def candlestick2_ohlc(ax, opens, highs, lows, closes, width=4,
     ax.autoscale_view()
 
     # add these last
-    ax.add_collection(barCollection)
     ax.add_collection(rangeCollection)
+    ax.add_collection(barCollection)
```

Comment:
every line should be covered by correlative bar, and its part between 'open' and 'close' should not been seen. As showed below:
![1 pic](https://cloud.githubusercontent.com/assets/17794885/17281122/201c9478-57cc-11e6-8c7b-52aaa369aac5.jpg)

logically, lines first, bars later. it's not enough, bars should be filled with color and cover lines. I'm just learning python here, so i don't know how to do a little further.